protocol/directory_test.go

```diff
@@ -138,3 +139,231 @@ func TestRegisterExistedUserWithoutTB(t *testing.T) {
+	if err != ErrorNameNotFound ||
+		res.Error != ErrorNameNotFound {
+		t.Fatal("Expect error code", ErrorNameNotFound, "got", err)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+}
+
+func TestKeyLookupWithoutTB(t *testing.T) {
+	d := newDirectory(t, false)
+	_, err := d.Register(&RegistrationRequest{
+		Username: "alice",
+		Key:      []byte("key")})
+	if err != Success {
+		t.Fatal("Unable to register")
+	}
+	// lookup in the same epoch
+	// expect a proof of absence
+	// and error ErrorNameNotFound
+	res, err := d.KeyLookup(&KeyLookupRequest{Username: "alice"})
+	if err != ErrorNameNotFound ||
+		res.Error != ErrorNameNotFound {
+		t.Fatal("Expect error code", ErrorNameNotFound, "got", err)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+
+	d.Update()
+	// lookup in epoch after registering epoch
+	// expect a proof of inclusion
+	res, err = d.KeyLookup(&KeyLookupRequest{Username: "alice"})
+	if ap := res.AP; ap == nil || !bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of inclusion")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+}
+
+func TestKeyLookupNameNotFoundWithoutTB(t *testing.T) {
+	d := newDirectory(t, false)
+	_, err := d.Register(&RegistrationRequest{
+		Username: "alice",
+		Key:      []byte("key")})
+	if err != Success {
+		t.Fatal("Unable to register")
+	}
+	// lookup in the same epoch
+	// expect a proof of absence
+	res, err := d.KeyLookup(&KeyLookupRequest{Username: "alice"})
+	if err != ErrorNameNotFound ||
+		res.Error != ErrorNameNotFound {
+		t.Fatal("Expect error code", ErrorNameNotFound, "got", err)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+
+	d.Update()
+	// lookup in different epochs
+	// expect a proof of absence
+	res, err = d.KeyLookup(&KeyLookupRequest{Username: "bob"})
+	if err != ErrorNameNotFound ||
+		res.Error != ErrorNameNotFound {
+		t.Fatal("Expect error code", ErrorNameNotFound, "got", err)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+}
+
+func TestConsistencyCheck(t *testing.T) {
+	N := 10
+
+	d := newDirectory(t, false)
+	_, err := d.Register(&RegistrationRequest{
+		Username: "alice",
+		Key:      []byte("key")})
+	if err != Success {
+		t.Fatal("Unable to register")
+	}
+
+	d.Update()
+	savedSTR := d.LatestSTR().Signature
+	for i := 2; i < N; i++ {
+		d.Update()
+	}
+
+	// missed from epoch 2
+	res, err := d.ConsistencyCheck(&ConsistencyCheckRequest{"alice", uint64(2), uint64(11)})
```

Comment:
I see, it definitely makes sense to test this. But it wasn't clear that this is what this part of the test is checking. I'm thinking this test should have two parts since checking that the server returns a valid hashchain given a range of epochs is maybe a different test than checking that the number of STRs returned is correct.